wrappers/python/examples/llama_training.py

```diff
@@ -103,114 +96,234 @@
+    pos_ids = np.repeat(single_batch_pos_ids, args.minibatch_size, axis=0)
+
+    mask = np.array(np.triu(np.ones((args.seq_len, args.seq_len))),
+                        dtype=bool, order="F")
+    llama_nntile, next_tag = Llama_nntile.from_torch(model_torch,
+                                                    args.minibatch_size,
+                                                    args.minibatch_size_tile,
+                                                    args.seq_len,
+                                                    args.seq_len_tile,
+                                                    pos_ids,
+                                                    mask,
+                                                    llama_config_nntile,
+                                                    next_tag)
+    time1 = time.time() - time0
+    print("Converting PyTorch model to NNTile",
+          "requires {} seconds".format(time1))
+    del model_torch
+
+if args.nntile_nepochs > 0:
+    from pathlib import Path
+
+    import numpy as np
+    from datasets import load_dataset
+    if args.dataset == "tinystories":
+        if args.dataset_file == args.dataset + "_train.bin":
+            train_data = np.memmap(Path(args.dataset_path) /
+                                    args.dataset_file,
+                                   dtype=np.uint16, mode='r')
+            train_tokens_raw = np.array(train_data, order='F', dtype=np.int64)
+            del train_data
+        elif args.dataset_file == "":
+            train_dataset = load_dataset("roneneldan/TinyStories",
+                            split='train', cache_dir=args.dataset_path)
+            if args.dataset_select != -1:
+                train_dataset = train_dataset.select(np.arange(
+                                args.dataset_select, dtype=np.int64))
+
+            map_train_tokens = map(lambda x: tokenizer(x["text"])["input_ids"],
+                                    train_dataset)
+            list_train_tokens = []
+            for seq in map_train_tokens:
+                list_train_tokens.extend(seq)
+            train_tokens_raw = np.array(list_train_tokens, dtype=np.int64)
+        else:
+            raise ValueError("Incompatible datasetfile")
+
+        num_train_tokens = train_tokens_raw.shape[0]
+
+        num_train_seq = num_train_tokens // (args.seq_len + 1)
+        num_train_batches = num_train_seq // args.batch_size
+        num_train_tokens_truncated = num_train_batches * (args.batch_size
+                * (args.seq_len + 1))
+        train_tokens_trunc = np.array(
+            train_tokens_raw[:num_train_tokens_truncated],
+            order='F', dtype=np.int64)
+        train_tokens = train_tokens_trunc.reshape(num_train_batches,
+                                            num_minibatch,
+                                            args.minibatch_size,
+                                            args.seq_len + 1)
+
+        if args.save_dataset_bin:
```

Comment:
So, this single script also saves data in a `train.bin` file. Why not make it a separate script? One script prepares any textual dataset from `load_dataset` into a `train.bin` file. Another script trains with the provided `train.bin`.